contracts/KwentaNFT.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@rari-capital/solmate/src/tokens/ERC1155.sol";
+
+error CallerIsNotOwner(address owner);
+error TokenIdOutOfRange(uint256 tokenId);
+error HasDistributed(bool hasDistributed);
+error MintIsDisabled(bool isMintingDisabled);
+
+contract KwentaNFT is ERC1155 {
+    using Strings for uint256;
+
+    // public state vars
+    address public immutable owner;
+    bool public hasDistributed;
+    bool public isMintDisabled;
+    // private state vars
+    string private baseURI;
+
+    constructor(string memory _baseURI) ERC1155() {
+        baseURI = _baseURI;
+        owner = msg.sender;
+    }
+
+    function uri(uint256 tokenId)
+        public
+        view
+        virtual
+        override
+        returns (string memory)
+    {
+        return
+            string(
+                abi.encodePacked(
+                    baseURI,
+                    getTierByTokenID(tokenId).toString(),
+                    ".json"
+                )
+            );
+    }
+
+    function getTierByTokenID(uint256 tokenId)
+        public
+        pure
+        returns (uint256 tierId)
+    {
+        if (tokenId < 101) return 0;
```

Comment:
One way to remove these hard-coded amounts would be to store a mapping of `tokenId -> tier`. It would require a different proposed change to the distribute function as well.